lib/process_win32.cpp

```diff
@@ -44,32 +68,30 @@ Number call(const utf8string &command, std::vector<unsigned char> *out, std::vec
     CloseHandle(err_write);
     out->clear();
     err->clear();
-    while (out_read != INVALID_HANDLE_VALUE || err_read != INVALID_HANDLE_VALUE) {
-        char buf[1024];
-        DWORD n;
-        if (out_read != INVALID_HANDLE_VALUE) {
-            if (not ReadFile(out_read, buf, sizeof(buf), &n, NULL)) {
-                CloseHandle(out_read);
-                out_read = INVALID_HANDLE_VALUE;
-                continue;
-            }
-            std::copy(buf, buf+n, std::back_inserter(*out));
-        }
-        if (err_read != INVALID_HANDLE_VALUE) {
-            if (not ReadFile(err_read, buf, sizeof(buf), &n, NULL)) {
-                CloseHandle(err_read);
-                err_read = INVALID_HANDLE_VALUE;
-                continue;
-            }
-            std::copy(buf, buf+n, std::back_inserter(*err));
-        }
-    }
+
+    ThreadInfo stdInfo[2];
+    stdInfo[0].pHandle = out_read;
+    stdInfo[0].buffer = out;
+    stdInfo[1].pHandle = err_read;
+    stdInfo[1].buffer = err;
+    DWORD threadId[2];
+    HANDLE hStdReaderHandle[2];
+
+    hStdReaderHandle[0] = CreateThread(NULL, 0, reader_thread, &stdInfo[0], 0, &threadId[0]);
+    hStdReaderHandle[1] = CreateThread(NULL, 0, reader_thread, &stdInfo[1], 0, &threadId[1]);
+
+    // Wait for the stdio readers to exit before waiting on the process to exit.
+    WaitForMultipleObjects(2, hStdReaderHandle, TRUE, INFINITE);
     if (out_read != INVALID_HANDLE_VALUE) {
```

Comment:
Don't need to check for invalid handle here.

Reply:
Fixed.